server/src/controllers/attainment.ts

```diff
@@ -313,3 +298,37 @@ export async function getAttainment(req: Request, res: Response): Promise<void>
     data: attainment,
   });
 }
+
+export async function getAllAttainments(req: Request, res: Response): Promise<void> {
+  const courseId: number = Number(req.params.courseId);
+  const instanceId: number = Number(req.params.instanceId);
+
+  await idSchema.validate({ id: courseId }, { abortEarly: false });
+  await idSchema.validate({ id: instanceId }, { abortEarly: false });
+  await treeSchema.validate(req.query, { abortEarly: false });
+
+  // Assert string type, as the query is validated above
+  const tree: string | undefined = req.query.tree as string | undefined;
```

Comment:
Ok, so no URL params, just get the full tree?

Reply:
Actually you can keep this here. In some cases there may be a need to get just the root attainment. If it turns out to be unnecessary this can be changed or removed later.